pkgs/tools/graphics/waifu2x-ncnn-vulkan/default.nix

```diff
@@ -0,0 +1,58 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, vulkan-headers
+, vulkan-loader
+, glslang
+, libgcc
+, libwebp
+, ncnn
+}:
+
+stdenv.mkDerivation rec {
+  pname = "waifu2x-ncnn-vulkan";
+  version = "20220419";
+
+  src = fetchFromGitHub {
+    owner = "nihui";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-cyPqbVEbASZdbUTHnlC8zDmDMiziwbEL9WDcCdN4GFo=";
+  };
+  sourceRoot = "source/src";
+
+  patches = [
+    # Fixes missing target SPIRV-Tools-opt
+    ./cmakelists.patch
+  ];
+
+  cmakeFlags = [
+    "-DUSE_SYSTEM_NCNN=1"
+    "-DUSE_SYSTEM_WEBP=1"
+    "-DGLSLANG_TARGET_DIR=${glslang}/lib/cmake"
+  ];
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ vulkan-headers vulkan-loader glslang libwebp ncnn ]
+    ++ lib.optional (!stdenv.isDarwin) libgcc;
+
+  installPhase = ''
+    runHook preInstall
+    # The models are expected to be next to the executable, so symlink the binary as to not pollute `$out/bin`.
```

Comment:
Alternatively, you can patch where it looks for models using

```
sed -i 's|path_t model = PATHSTR("models-cunet")|path_t model = PATHSTR("'$out'/models/models-cunet")|' src/main.cpp
```

in `postPatch`

Reply:
This does seem like a cleaner solution, although I'd vouch for using `substituteInPlace` over `sed` seeing how this is a simple find-and-replace, the former is a lot more readable for that.